src/ralph/models/edx/navigational.py

```diff
@@ -0,0 +1,140 @@
+"""Navigational event model definitions"""
+
+from typing import Literal, Union
+
+from pydantic import Json, validator
+
+from ralph.models.selector import selector
+
+from .base import BaseEventFieldModel
+from .browser import BaseBrowserEventModel
+
+
+class PageCloseBrowserEventModel(BaseBrowserEventModel):
+    """Represents the `page_close` browser event.
+
+    This type of event is triggered when the user navigates to the next page
+    or closes the browser window (when the JavaScript `window.onunload` event
+    is called).
+
+    Attributes:
+        name (str): Consists of the value `page_close`.
+        event_type (str): Consists of the value `page_close`.
+        event (str): Consists of the string value `{}`.
+    """
+
+    __selector__ = selector(event_source="browser", event_type="page_close")
+
+    # pylint: disable=unsubscriptable-object
+    name: Literal["page_close"]
+    event_type: Literal["page_close"]
+    event: Literal["{}"]
+
+
+class NavigationalBrowserEventEventFieldModel(BaseEventFieldModel):
+    """Represents the event field of navigational events.
+
+    Note: All navigational events are `browser` events.
+
+    Attributes:
+        id (int): Consists of the edX ID of the sequence.
+        old (str): For `seq_goto`, it consists of the index of the unit being jumped to.
+            For `seq_next` and `seq_prev`, it consists of the index of the unit being navigated to.
+        new (str): For `seq_goto`, it consists of the index of the unit being jumped from.
+            For `seq_next` and `seq_prev`, it consists of the index of the unit being navigated
+        away from.
+    """
+
+    id: str  # FIXME with string regex
```

Comment:
That's a good idea. I have tried to implement like in your marshmallow implementation so that we can have also course_key and block_id in the instance. Tests are still failing but it should be ready this afternoon